addOns/scripts/src/main/java/org/zaproxy/zap/extension/scripts/automation/ui/ScriptJobDialog.java

```diff
@@ -56,42 +73,114 @@ public ScriptJobDialog(ScriptJob job) {
+        for (String fieldName : ALL_FIELDS) {
+            if (disabledFields.contains(fieldName)) {
+                this.setFieldValue(fieldName, "");
+                this.getField(fieldName).setEnabled(false);
+            } else {
+                this.getField(fieldName).setEnabled(true);
+            }
+        }
     }
 
     private void onScriptTypeChanged() {
         String scriptType = this.getStringValue(SCRIPT_TYPE_PARAM);
         List<String> scripts = ScriptAction.getAvailableScriptNames(scriptType);
+        // Always have a blank option at the start
+        scripts.add(0, "");
         this.setComboFields(
                 SCRIPT_NAME_PARAM, scripts, this.job.getData().getParameters().getName());
     }
 
+    private void onScriptNameChanged() {
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            String scriptName = this.getStringValue(SCRIPT_NAME_PARAM);
+            ScriptWrapper sw = ScriptAction.getExtScript().getScript(scriptName);
+            if (sw != null) {
+                this.setFieldValue(SCRIPT_FILE_PARAM, sw.getFile().getAbsolutePath());
+            }
+        }
+    }
+
     @Override
     public void save() {
         this.job.getData().setName(this.getStringValue(NAME_PARAM));
         this.job.getData().getParameters().setAction(this.getStringValue(SCRIPT_ACTION_PARAM));
         this.job.getData().getParameters().setType(this.getStringValue(SCRIPT_TYPE_PARAM));
+        this.job.getData().getParameters().setEngine(this.getStringValue(SCRIPT_ENGINE_PARAM));
         this.job.getData().getParameters().setName(this.getStringValue(SCRIPT_NAME_PARAM));
+
+        ScriptAction sa = getScriptAction();
+        if (sa.getDisabledFields().contains(SCRIPT_FILE_PARAM)) {
+            this.job.getData().getParameters().setFile(null);
+        } else {
+            File f = new File(this.getStringValue(SCRIPT_FILE_PARAM));
+            if (f.isFile()) {
+                this.job.getData().getParameters().setFile(this.getStringValue(SCRIPT_FILE_PARAM));
+            }
+        }
         this.job.resetAndSetChanged();
     }
 
     @Override
     public String validateFields() {
-        return null;
+        ScriptJobParameters params =
+                new ScriptJobParameters(
+                        this.getStringValue(SCRIPT_ACTION_PARAM),
+                        this.getStringValue(SCRIPT_TYPE_PARAM),
+                        this.getStringValue(SCRIPT_ENGINE_PARAM),
+                        this.getStringValue(SCRIPT_NAME_PARAM),
+                        this.getStringValue(SCRIPT_FILE_PARAM));
+        ScriptAction sa = ScriptJob.createScriptAction(params, null);
+        List<String> issues = sa.verifyParameters(this.job.getName(), params, null);
+        if (issues.isEmpty()) {
+            // No problems
+            return null;
+        }
+        return issues.stream().collect(Collectors.joining("\n"));
```

Comment:
on Warnings like
"File specified but not needed"

Save is not done (Popup stays open)

Reply:
Thats kind of what I want, but then its not possible to edit the field as its disabled :/
Some tweeking is required then...